vividus-plugin-rest-api/src/main/java/org/vividus/http/HttpRequestExecuter.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.http;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.client.IHttpClient;
+
+public class HttpRequestExecuter
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpRequestExecuter.class);
+
+    private IHttpClient httpClient;
+    @Inject private HttpTestContext httpTestContext;
+
+    public HttpResponse execute(HttpMethod httpMethod, String endpoint, String relativeURL) throws IOException
+    {
+        HttpRequestBase requestBase = prepareHttpRequestBase(httpMethod, endpoint, relativeURL);
+        HttpClientContext httpClientContext = new HttpClientContext();
+        httpTestContext.getCookieStore().ifPresent(httpClientContext::setCookieStore);
+        httpTestContext.getRequestConfig().ifPresent(httpClientContext::setRequestConfig);
+
+        HttpResponse httpResponse = httpClient.execute(requestBase, httpClientContext);
+        LOGGER.info("Response time: {} ms", httpResponse.getResponseTimeInMs());
+        return httpResponse;
+    }
+
+    private HttpRequestBase prepareHttpRequestBase(HttpMethod httpMethod, String endpoint, String relativeURL)
+    {
+        if (httpTestContext.getRequestBase() == null)
+        {
+            HttpRequestBuilder httpRequestBuilder = HttpRequestBuilder.create()
+                    .withHttpMethod(httpMethod)
+                    .withEndpoint(endpoint)
+                    .withRelativeUrl(relativeURL)
+                    .withHeaders(httpTestContext.pullRequestHeaders());
+            httpTestContext.pullRequestEntity().ifPresent(httpRequestBuilder::withContent);
+            httpTestContext.putRequestBase(httpRequestBuilder.build());
```

Comment:
HTTP request object can be used only one time, since it keeps HTTP execution details